java/src/main/java/com/powsybl/python/SensitivityAnalysisContext.java

```diff
@@ -28,26 +29,107 @@
 
     private List<SensitivityVariableSet> variableSets = Collections.emptyList();
 
-    private List<String> branchFlowBranchesIds;
+    static class MatrixInfo {
+        private final ContingencyContextType contingencyContextType;
 
-    private List<String> branchFlowVariablesIds;
+        private final SensitivityFunctionType functionType;
 
-    private List<String> busVoltageEquipmentsIds;
+        private final List<String> columnIds;
 
-    private List<String> targetVoltageEquipmentsIds;
+        private final List<String> rowIds;
+
+        private final List<String> contingencyIds;
+
+        private int offsetData;
+
+        private int offsetColumn;
+
+        MatrixInfo(ContingencyContextType context, SensitivityFunctionType functionType, List<String> columnIds, List<String> rowIds, List<String> contingencyIds) {
+            this.contingencyContextType = context;
+            this.functionType = functionType;
+            this.columnIds = columnIds;
+            this.rowIds = rowIds;
+            this.contingencyIds = contingencyIds;
+        }
+
+        ContingencyContextType getContingencyContextType() {
+            return contingencyContextType;
+        }
+
+        SensitivityFunctionType getFunctionType() {
+            return functionType;
+        }
+
+        void setOffsetData(int offset) {
+            this.offsetData = offset;
+        }
+
+        void setOffsetColumn(int offset) {
+            this.offsetColumn = offset;
+        }
+
+        int getOffsetData() {
+            return offsetData;
+        }
+
+        int getOffsetColumn() {
+            return offsetColumn;
+        }
+
+        List<String> getRowIds() {
+            return rowIds;
+        }
+
+        List<String> getColumnIds() {
+            return columnIds;
+        }
+
+        List<String> getContingencyIds() {
+            return contingencyIds;
+        }
+
+        int getRowCount() {
+            return rowIds.size();
+        }
+
+        int getColumnCount() {
+            return columnIds.size();
+        }
+    }
+
+    private final Map<String, MatrixInfo> branchFlowFactorsMatrix = new HashMap<>();
+
+    private MatrixInfo busVoltageFactorsMatrix;
+
+    private final Map<String, MatrixInfo> preContingencyBranchFlowFactorMatrix = new HashMap<>();
```

Comment:
Why do we need 3 different maps branchFlowFactorsMatrix, preContingencyBranchFlowFactorMatrix, postContingencyBranchFlowFactorMatrix ?

Reply:
Matrices are put in a unique container in prepareMatrices method. I cannot see any reason for the separated maps. There may have been one earlier but I can't remember. This will be cleaned.